src/combinatorics/additive/behrend.lean

```diff
@@ -196,4 +199,341 @@ begin
+    have : ((12 : ℕ) : ℝ) * log 2 ≤ log N,
+    { rw [←log_rpow zero_lt_two, log_le_log, rpow_nat_cast],
+      { norm_num1,
+        exact_mod_cast hN },
+      { exact rpow_pos_of_pos zero_lt_two _ },
+      rw cast_pos,
+      exact hN.trans_lt' (by norm_num1) },
+    refine le_trans _ this,
+    simp only [cast_bit0, cast_bit1, cast_one],
+    rw ←div_le_iff',
+    { exact log_two_gt_d9.le.trans' (by norm_num1) },
+    { norm_num1 } },
+  { rw cast_pos,
+    exact hN.trans_lt' (by norm_num1) },
+  { refine div_pos zero_lt_two _,
+    rw [sub_pos, div_lt_one (exp_pos _)],
+    exact lt_of_le_of_lt (by norm_num1) exp_one_gt_d9 },
+  apply rpow_pos_of_pos,
+  rw cast_pos,
+  exact hN.trans_lt' (by norm_num1),
+end
+
+lemma roth_lower_bound_explicit (hN : 4096 ≤ N) :
+  (N : ℝ) * exp (-4 * sqrt (log N)) < roth_number_nat N :=
+begin
+  let n := n_value N,
+  have hn : 0 < (n : ℝ) := cast_pos.2 (n_value_pos $ hN.trans' $ by norm_num1),
+  have hd : 0 < d_value N := d_value_pos (hN.trans' $ by norm_num1),
+  have hN₀ : 0 < (N : ℝ) := cast_pos.2 (hN.trans' $ by norm_num1),
+  have hn₂ : 2 ≤ n := two_le_n_value (hN.trans' $ by norm_num1),
+  have : (2 * d_value N - 1)^n ≤ N := le_N (hN.trans' $ by norm_num1),
+  refine ((bound_aux hd.ne' hn₂).trans $ cast_le.2 $ roth_number_nat.mono this).trans_lt' _,
+  refine (div_lt_div_of_lt hn $ pow_lt_pow_of_lt_left (bound hN) _ _).trans_le' _,
+  { exact div_nonneg (rpow_nonneg_of_nonneg (cast_nonneg _) _) (exp_pos _).le },
+  { exact tsub_pos_of_lt (three_le_n_value $ hN.trans' $ by norm_num1) },
+  rw [←rpow_nat_cast, div_rpow (rpow_nonneg_of_nonneg hN₀.le _) (exp_pos _).le, ←rpow_mul hN₀.le,
+    mul_comm (_ / _), mul_one_div, cast_sub hn₂, cast_two, same_sub_div hn.ne', exp_one_rpow,
+    div_div, rpow_sub hN₀, rpow_one, div_div, div_eq_mul_inv],
+  refine mul_le_mul_of_nonneg_left _ (cast_nonneg _),
+  rw [mul_inv, mul_inv, ←exp_neg, ←rpow_neg (cast_nonneg _), neg_sub, ←div_eq_mul_inv],
+  have : exp ((-4) * sqrt (log N)) = exp (-2 * sqrt (log N)) * exp (-2 * sqrt (log N)),
+  { rw [←exp_add, ←add_mul],
+    norm_num },
+  rw this,
+  refine (mul_le_mul _ (exp_neg_two_mul_le $ real.sqrt_pos.2 $ log_pos _).le (exp_pos _).le $
+    rpow_nonneg_of_nonneg (cast_nonneg _) _),
+  { rw [←le_log_iff_exp_le (rpow_pos_of_pos hN₀ _), log_rpow hN₀, ←le_div_iff, mul_div_assoc,
+      div_sqrt, neg_mul, neg_le_neg_iff, div_mul_eq_mul_div, div_le_iff hn],
+    { exact mul_le_mul_of_nonneg_left (le_ceil _) zero_le_two },
+    refine real.sqrt_pos.2 (log_pos _),
+    rw one_lt_cast,
+    exact hN.trans_lt' (by norm_num1) },
+  { rw one_lt_cast,
+    exact hN.trans_lt' (by norm_num1) }
+end
+
+lemma exp_four_lt : exp 4 < 64 :=
+begin
+  rw [show (64 : ℝ) = 2 ^ ((6 : ℕ) : ℝ), by norm_num1,
+    ←lt_log_iff_exp_lt (rpow_pos_of_pos zero_lt_two _), log_rpow zero_lt_two, ←div_lt_iff'],
+  exact log_two_gt_d9.trans_le' (by norm_num1),
+  norm_num
+end
+
+lemma four_zero_nine_six_lt_exp_sixteen : 4096 < exp 16 :=
```

Comment:
Can names start with a digit? If not, maybe just prepend a `_`, since it's an aux-lemma anyway.
I think `_4096_lt_log_16` is a nicer name (-;